packages/blockchain-link-utils/src/solana.ts

```diff
@@ -94,12 +124,33 @@ export const getTxType = (
         return 'unknown';
     }
 
+    const isInstructionCreatingTokenAccount = (instruction: ParsedInstruction) =>
+        instruction.program === 'spl-associated-token-account' &&
+        instruction.parsed.type === 'create';
+
     const isTransfer = parsedInstructions.every(
-        instruction => instruction.parsed.type === 'transfer',
+        instruction =>
+            instruction.parsed.type === 'transfer' ||
+            instruction.parsed.type === 'transferChecked' ||
+            isInstructionCreatingTokenAccount(instruction),
     );
 
     // for now we support only transfers, so we interpret all other transactions as `unknown`
     if (isTransfer) {
+        // Token Transfers
+        if (tokenEffects.length > 0) {
+            if (tokenEffects.every(({ type }) => type === 'recv')) {
+                return 'recv';
+            }
+
+            if (tokenEffects.every(({ type }) => type === 'sent')) {
+                return 'sent';
+            }
+
+            return 'unknown';
+        }
```

Comment:
NP: Could we maybe split this into `isNativeTransfer` and `isTokenTransfer`, and avoid this long nested `ifs` ?